src/Http/Http.Extensions/gen/StaticRouteHandlerModel/InvocationOperationExtensions.cs

```diff
@@ -11,52 +11,91 @@ namespace Microsoft.AspNetCore.Http.RequestDelegateGenerator.StaticRouteHandlerM
 internal static class InvocationOperationExtensions
 {
     private const int RoutePatternArgumentOrdinal = 1;
-    private const int RouteHandlerArgumentOrdinal = 2;
+    public static readonly string[] KnownMethods =
+    {
+        "MapGet",
+        "MapPost",
+        "MapPut",
+        "MapDelete",
+        "MapPatch",
+        "Map",
+        "MapMethods",
+        "MapFallback"
+    };
 
     public static bool TryGetRouteHandlerMethod(this IInvocationOperation invocation, SemanticModel semanticModel, out IMethodSymbol? method)
     {
+        method = null;
+        if (invocation.TryGetRouteHandlerArgument(out var argument))
+        {
+            method = ResolveMethodFromOperation(argument, semanticModel);
+            return true;
+        }
+        return false;
+    }
+
+    public static bool TryGetRouteHandlerArgument(this IInvocationOperation invocation, [NotNullWhen(true)]out IArgumentOperation? argumentOperation)
+    {
+        argumentOperation = null;
+        // Route handler argument is typically the last parameter provided to
+        // the Map methods.
```

Comment:
> is typically the last parameter

What happens if it isn't the last parameter?

Reply:
For all of the overloads that we currently support, the handler parameter is the last one. If we did end up designing an API that didn't take it as the last one, we'd have to explicitly support that and we'd have the ability to change this code.

But it would mean that we designed an API that users had to invoke like this:

```
app.NewMap("/", (int id) => "Hello world!", new[] { "GET" });
```

Which be particularly unusual. Virtual all APIs that take a delegate take it as the last parameter and I would be surprised if we saw us step away from that pattern. I think this is a safe assumption to encode in the code.

Reply:
Maybe just updating the comment then:

```suggestion
        // Route handler argument is assumed to be the last parameter provided to
        // the Map methods.
```